### DISCUSSION TITLE
Segfault during bootstrap

### DISCUSSION THREAD

**Discussion by adrianlshaw** [toltec-dev/toltec#616] [Support] [answered]
Hi all. I'm trying to install toltec from the site instructions (https://toltec-dev.org/ But when I run bootstrap I get the following error

```sh
reMarkable: ~/ ./bootstrap 
INFO:  Bootstrapping Opkg
./bootstrap: line 119:   575 Segmentation fault      (core dumped) wget --no-verbose "$entware_remote/opkg" --output-document "$opkg_path"
ERROR: Could not fetch opkg, make sure you have a stable Wi-Fi connection
```

This is a Remarkable 1.

**Comment by matteodelabre** [comment]
Hello Adrian, thanks for the report. Are you able to reproduce this consistently with every bootstrap run? Did you check that your bootstrap script has the correct checksum?

I don’t think this is what’s causing the segfault, but the bootstrap script uses some Bash constructs, so you’ll need to run `bash bootstrap` unless you have changed your default shell.

**Reply by adrianlshaw** [reply]
Yes, I can reproduce it again. The checksum appears to be valid. See below, you can see it runs in bash.

```bash
reMarkable: ~/ rm bootstrap 
reMarkable: ~/ wget http://toltec-dev.org/bootstrap
Connecting to toltec-dev.org (5.161.99.157:80)
bootstrap            100% |************************************************|  8491   0:00:00 ETA
reMarkable: ~/ echo "04a28483286f88c5c7f39e352afb62adc57f6162a29fd7e124d832205bb0980e  bootstrap" | sha256sum -c && bash bootstrap
bootstrap: OK
INFO:  Bootstrapping Opkg
bootstrap: line 119:   722 Segmentation fault      (core dumped) wget --no-verbose "$entware_remote/opkg" --output-document "$opkg_path"
ERROR: Could not fetch opkg, make sure you have a stable Wi-Fi connection
```

**Reply by matteodelabre** [reply]
Thanks for confirming. It looks like this is the default wget binary shipped with the reMarkable system which is causing the segfault. Could you share the OS version you’re running this from?

**Reply by adrianlshaw** [reply]
20190228165944

**Comment by adrianlshaw** [comment, marked as answer]
I'll bite the bullet and update the OS as per @matteodelabre's advice.
